pycbc/waveform/bank.py

```diff
@@ -329,12 +329,12 @@ def ensure_hash(self):
 
         fields = [f for f in hash_fields if f in fields]
         template_hash = numpy.array([hash(v) for v in zip(*[self.table[p]
-                         for p in fields])]) 
+                         for p in fields])])
         self.table = self.table.add_fields(template_hash, 'template_hash')
 
-    def write_to_hdf(self, filename, force=False, skip_fields=None):
+    def write_to_hdf(self, filename, start_index, stop_index, force=False, skip_fields=None):
```

Comment:
Make `start_index` and `stop_index` keyword arguments, with the default for each set to None. That will cause it to write all templates by default.